src/components/offers/createForm/OfferForm.js

```diff
@@ -13,39 +14,64 @@ const useStyles = makeStyles((theme) => ({
 }))
 
 export const OfferForm = (props) => {
+    const { id } = useParams()
+
+
     const defaultFormState = {
-        name:"",
-        position:"",
-        localization:"",
-        minSalary:"",
-        maxSalary:"",
-        description:"",
-        creatorId:"",
-        skills:[]
+        name: "",
+        position: "",
+        localization: "",
+        minSalary: "",
+        maxSalary: "",
+        description: "",
+        creatorId: "",
+        skills: []
     }
+
+
+
     const {handleSubmit, watch, trigger, control, reset} = useForm({mode: "onChange", defaultValues:defaultFormState})
 
     const minSalary = watch("minSalary", null)
     const maxSalary = watch("maxSalary", null)
 
+    function updateOffer(data) {
+        defaultFormState.name = data.name
+        defaultFormState.position = data.position
+        defaultFormState.localization = data.localization
+        defaultFormState.minSalary = data.minSalary
+        defaultFormState.maxSalary = data.maxSalary
+        defaultFormState.description = data.description
+        defaultFormState.skills = data.skills
+        reset(defaultFormState)
+    }
+
+    useEffect(() => {
+        if(id !== undefined){
+            offersAPI.getOfferById(id)
+                .then(data => updateOffer(data))
+        }
+    }, [id])
+
     const classes = useStyles()
 
     useEffect(() => {if(maxSalary) trigger("minSalary")}, [trigger, maxSalary])
     useEffect(() => {if(minSalary) trigger("maxSalary")}, [trigger, minSalary])
 
     const onSubmit = (data) => {
-        const formResult = Object.assign(data, {
+        let formResult = Object.assign(data, {
```

Comment:
let się tu znalazł przez pomyłkę, wywalę jak przyjdzie więcej poprawek XD

Reply:
Spoko, let też nie jest zły XDD Ale to jak z val i var w Kotlinie - niby można, ale lepiej mieć niemodyfikowalne zmienne 😄